lib/Dialect/Emit/EmitOps.cpp

```diff
@@ -0,0 +1,58 @@
+//===- EmitOps.cpp - Implement the Emit operations ------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements `sim` dialect ops.
+//
+//===----------------------------------------------------------------------===//
+
+#include "circt/Dialect/Emit/EmitOps.h"
+
+using namespace mlir;
+using namespace circt;
+using namespace emit;
+
+//===----------------------------------------------------------------------===//
+// FileOp
+//===----------------------------------------------------------------------===//
+
+void FileOp::build(OpBuilder &builder, OperationState &result,
+                   StringRef symName, StringRef fileName,
+                   llvm::function_ref<void()> bodyCtor) {
+  MLIRContext *context = builder.getContext();
+  OpBuilder::InsertionGuard guard(builder);
+
+  result.addAttribute("sym_name", StringAttr::get(context, symName));
+  result.addAttribute("file_name", StringAttr::get(context, fileName));
+
+  builder.createBlock(result.addRegion());
+  if (bodyCtor)
+    bodyCtor();
+}
+
+//===----------------------------------------------------------------------===//
+// FileListOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult FileListOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
+  auto *parent = getOperation()->getParentOp();
+  for (auto sym : getFiles()) {
+    Operation *op =
+        symbolTable.lookupSymbolIn(parent, sym.cast<FlatSymbolRefAttr>());
```

Comment:
It looks like FileOp and FileListOp are expected to be siblings that exist within the same level of SymbolTable. Should we write that down in the ODS description? Could we go farther and maybe use HasParent with the same kind of op for both? Or do we not care, as long as their siblings?

Reply:
HasParent is tricky since at the FIRRTL level these should live inside of a CircuitOp, but at the HW level they live inside a ModuleOp.

Reply:
Makes sense. I guess we should just document in FileListOp that it expects the FileOps it references to be in the same symbol table as the FileListOp itself.

Reply:
It should fail if it's not.  The lookup should be structure to ensure that, which might mean not poking at random op parents to do lookup but following the symbol lookup rules.

Reply:
This looks good to me now using `lookupNearestSymbolFrom(getOperation(), ...)`